developer/src/tike/child/UfrmDebug.pas

```diff
@@ -448,19 +449,23 @@ function TfrmDebug.ProcessKeyEvent(var Message: TMessage): Boolean;
     end;
   end;
 var
-  modifier: uint16_t;
+  vkey, modifier: uint16_t;
 begin
   Assert(Assigned(FDebugCore));
+
+  // We always use the US virtual key code as a basis for our keystroke
+  // mapping; the best way to do this is to extract the scan code from
+  // the message data and work from that
+  vkey := MapScanCodeToUSVK((Message.LParam and $FF0000) shr 16);
```

Comment:
Since the debugger is processing scan codes, does this mean the user testing involves keyboard hardware (e.g. a physical AZERTY keyboard)?

Reply:
An AZERTY keyboard produces identical scan codes to a QWERTY keyboard. There are two differences: the extra key (D00) to right of the left shift key (D99), and the movement of B13 (backslash on US English) from ANSI/QWERTY to A13, to the left of backspace.

It is fine to test on US English hardware, so long as we select the French AZERTY keyboard in Windows language preferences.